base/regex.jl

```diff
@@ -152,8 +152,10 @@ function match(re::Regex, str::UTF8String, idx::Integer, add_opts::UInt32=UInt32
     RegexMatch(mat, cap, ovec[1]+1, off, re)
 end
 
+_utf8(str) = utf8(str)
```

Comment:
@StefanKarpinski It is actually possible to build a version of PCRE that supports utf8, utf16, and utf32. It would take some munging of pcre.jl to expose it to Julia - more than I'm willing to do, but that could be a decent intro issue.